Bottom: c2f03a0550b4cd8b00bc5a4ff407c8b82a441919
Top:    cc26e3bcdd01b377f10e0b1800340c2d21f8307b
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-22 16:52:14 -0400

Register the wm9712 DAIs on module load


---

diff --git a/sound/soc/codecs/wm9712.c b/sound/soc/codecs/wm9712.c
index 1fd4e88..49ad987 100644
--- a/sound/soc/codecs/wm9712.c
+++ b/sound/soc/codecs/wm9712.c
@@ -742,6 +742,18 @@ struct snd_soc_codec_device soc_codec_dev_wm9712 = {
 };
 EXPORT_SYMBOL_GPL(soc_codec_dev_wm9712);
 
+static int __init wm9712_modinit(void)
+{
+	return snd_soc_register_dais(wm9712_dai, ARRAY_SIZE(wm9712_dai));
+}
+module_init(wm9712_modinit);
+
+static void __exit wm9712_exit(void)
+{
+	snd_soc_unregister_dais(wm9712_dai, ARRAY_SIZE(wm9712_dai));
+}
+module_exit(wm9712_exit);
+
 MODULE_DESCRIPTION("ASoC WM9711/WM9712 driver");
 MODULE_AUTHOR("Liam Girdwood");
 MODULE_LICENSE("GPL");
diff --git a/sound/soc/fsl/mpc5200_psc_ac97.c b/sound/soc/fsl/mpc5200_psc_ac97.c
index 00e832f..ba8e114 100644
--- a/sound/soc/fsl/mpc5200_psc_ac97.c
+++ b/sound/soc/fsl/mpc5200_psc_ac97.c
@@ -322,6 +322,9 @@ static int __devinit psc_ac97_of_probe(struct of_device *op,
 	if (rc != 0)
 		return rc;
 
+	for (i = 0; i < ARRAY_SIZE(psc_ac97_dai); i++)
+		psc_ac97_dai[i].dev = &op->dev;
+
 	rc = snd_soc_register_dais(psc_ac97_dai, ARRAY_SIZE(psc_ac97_dai));
 	if (rc != 0) {
 		pr_err("Failed to register DAI\n");
diff --git a/sound/soc/fsl/pcm030-audio-fabric.c b/sound/soc/fsl/pcm030-audio-fabric.c
index 83ebfa5..4bd957a 100644
--- a/sound/soc/fsl/pcm030-audio-fabric.c
+++ b/sound/soc/fsl/pcm030-audio-fabric.c
@@ -55,8 +55,6 @@ static __init int pcm030_fabric_init(void)
 	if (!machine_is_compatible("phytec,pcm030"))
 		return -ENODEV;
 
-
-	printk("Initializing Phytec audio\n");
 	card.platform = &mpc5200_audio_dma_platform;
 	card.name = "pcm030";
 	card.dai_link = pcm030_fabric_dai;
